src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImpl.java

```diff
@@ -0,0 +1,621 @@
+//        
+//        vendorContacts.add(buildContact(PaymentWorksConstants.KFSVendorContactTypes.E_INVOICING, PaymentWorksConstants.KFSVendorContactPhoneTypes.E_INVOICING,
+//                                        pmwVendor.geteInvoiceContactName(), pmwVendor.geteInvoiceEmail(), 
+//                                        pmwVendor.geteInvoiceContactPhoneNumber(), pmwVendor.geteInvoicePhoneExtension()));
+//        
+//        vendorContacts.add(buildContact(PaymentWorksConstants.KFSVendorContactTypes.INSURANCE, PaymentWorksConstants.KFSVendorContactPhoneTypes.INSURANCE,
+//                                        pmwVendor.getInsuranceContactName(), pmwVendor.getInsuranceContactEmail(), 
+//                                        pmwVendor.getInsuranceContactPhoneNumber(), pmwVendor.getInsuranceContactPhoneExtension()));
+//
+//        vendorContacts.add(buildContact(PaymentWorksConstants.KFSVendorContactTypes.SALES, PaymentWorksConstants.KFSVendorContactPhoneTypes.SALES,
+//                                        pmwVendor.getSalesContactName(), pmwVendor.getSalesContactEmail(), 
+//                                        pmwVendor.getSalesContactPhoneNumber(), pmwVendor.getSalesContactPhoneExtension()));
+        return vendorContacts;
+    }
+    
+    private VendorContact buildContact(String contactType, String contactPhoneType, String contactName, String contactEmailAddress, String contactPhoneNumber, String contactPhoneExtension) {
+        List<VendorContactPhoneNumber> vendorContactPhoneNumbers = new ArrayList<VendorContactPhoneNumber>();
+        vendorContactPhoneNumbers.add(buildContactPhoneNumber(contactPhoneType, contactPhoneNumber, contactPhoneExtension));
+        VendorContact contact = new VendorContact();
+        contact.setVendorContactPhoneNumbers(vendorContactPhoneNumbers);
+        contact.setVendorContactTypeCode(contactPhoneType);
+        contact.setVendorContactName(contactName);
+        contact.setVendorContactEmailAddress(contactEmailAddress);
+        contact.setActive(true);
+        return contact;
+    }
+    
+    private VendorContactPhoneNumber buildContactPhoneNumber(String contactPhoneType, String contactPhoneNumber, String contactPhoneExtension) {
+        VendorContactPhoneNumber vendorContactPhoneNumber = new VendorContactPhoneNumber();
+        vendorContactPhoneNumber.setVendorPhoneTypeCode(contactPhoneType);
+        vendorContactPhoneNumber.setVendorPhoneNumber(contactPhoneNumber);
+        if (StringUtils.isNotBlank(contactPhoneExtension)) {
+            vendorContactPhoneNumber.setVendorPhoneExtensionNumber(contactPhoneExtension);
+        }
+        vendorContactPhoneNumber.setActive(true);
+        return vendorContactPhoneNumber;
+    }
+
+    private int determineTaxRuleToUseForDataPopulation(PaymentWorksVendor pmwVendor, String pmwVendorFipsTaxCountryCode) {
+        if (isPmwVendorIndividualSolePropriatorSingleMemberLlc(pmwVendor) && 
```

Comment:
perhaps the checks could be extracted into a method called isIndividualUsSsn? or something along those lines? same goes for the other checks in that determine the tax rule.

Reply:
Extracted each check into its own method.